core/src/crds_gossip_pull.rs

```diff
@@ -231,19 +238,17 @@ impl CrdsGossipPull {
         self.filter_crds_values(crds, requests)
     }
 
-    /// process a pull response
-    pub fn process_pull_response(
-        &mut self,
-        crds: &mut Crds,
-        from: &Pubkey,
+    pub fn filter_pull_responses(
+        &self,
+        crds: &Crds,
         timeouts: &HashMap<Pubkey, u64>,
-        response: Vec<CrdsValue>,
+        responses: Vec<CrdsValue>,
         now: u64,
-    ) -> (usize, usize, usize) {
-        let mut failed = 0;
-        let mut timeout_count = 0;
-        let mut success = 0;
-        for r in response {
+        stats: &mut ProcessPullStats,
+    ) -> (Vec<VersionedCrdsValue>, Vec<VersionedCrdsValue>) {
+        let mut versioned = vec![];
+        let mut versioned_no_timestamp = vec![];
```

Comment:
is `versioned_no_timestamp` all the expired/invalid timestamp labels? Maybe a better name would be `versioned_expired_timestamp`?